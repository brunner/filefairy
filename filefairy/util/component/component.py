#!/usr/bin/env python3
# -*- coding: utf-8 -*-


def anchor(url, content):
    return '<a href="{}">{}</a>'.format(url, content)


def card(href='', title='', info='', table=None, ts='', success='', danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }


def span(classes, text):
    return '<span class="{}">{}</span>'.format(' '.join(classes), text)


def table(clazz='border mt-3', hcols=None, bcols=None, head=None, body=None):
    return {
        'clazz': clazz,
        'hcols': hcols,
        'bcols': bcols,
        'head': head,
        'body': body
    }
