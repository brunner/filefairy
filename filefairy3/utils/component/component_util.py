#!/usr/bin/env python


def card(href='',
         title='',
         info='',
         table=[],
         ts='',
         success='',
         danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }
