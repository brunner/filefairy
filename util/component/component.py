#!/usr/bin/env python3
# -*- coding: utf-8 -*-


def anchor(url, content):
    return '<a href="{}">{}</a>'.format(url, content)


def bold(text):
    return span(['font-weight-bold'], text)


def card(href='',
         title='',
         info='',
         code='',
         table=None,
         ts='',
         success='',
         danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'code': code,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }


def cell(col=None, content=''):
    return {'col': col, 'content': content}


def col(clazz='', colspan=''):
    return {'clazz': clazz, 'colspan': colspan}


def show(trigger, text):
    s = '<a data-toggle="collapse" href="#{0}" role="button" ' + \
        'aria-expanded="false" aria-controls="{0}">{1}</a>'
    return s.format(trigger, text)


def replace(text, replace):
    s = '<a class="repl-toggler" href="#" role="button" data-repl="{}">{}</a>'
    return s.format(replace, text)


def secondary(text):
    return span(['text-secondary'], text)


def span(classes, text):
    return '<span class="{}">{}</span>'.format(' '.join(classes), text)


def table(clazz='border mt-3',
          id_='',
          hcols=None,
          bcols=None,
          fcols=None,
          head=None,
          body=None,
          foot=None):
    return {
        'clazz': clazz,
        'id': id_,
        'hcols': hcols,
        'bcols': bcols,
        'fcols': fcols,
        'head': head,
        'body': body,
        'foot': foot
    }
