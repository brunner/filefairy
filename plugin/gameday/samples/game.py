#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import re
import sys

_path = os.path.dirname(os.path.abspath(__file__))
sys.path.append(re.sub(r'/plugin/gameday', '', _path))
from util.component.component import anchor  # noqa
from util.component.component import replace  # noqa
from util.component.component import show  # noqa
from util.component.component import span  # noqa
from util.component.component import table  # noqa

_breadcrumbs = [{
    'href': '/',
    'name': 'Fairylab'
}, {
    'href': '/gameday/',
    'name': 'Gameday'
}, {
    'href': '',
    'name': 'Diamondbacks at Dodgers, 10/09/2022'
}]

_raw = [
    table(
        clazz='border mt-3',
        bcols=[' class="w-50"', ' class="w-50"'],
        hcols=[' colspan="2"'],
        fcols=[' colspan="2"'],
        head=[
            'Arizona Diamondbacks batting - Pitching for Los Angeles ' +
            'Dodgers : LHP 101'
        ],
        body=[['Batting: SHB 102', '0-0: Ball'], ['', '103 to second']],
        foot=['0 run(s), 0 hit(s), 0 error(s), 0 left on base'])
]

_schedule = [
    table(
        clazz='table-fixed border border-bottom-0',
        hcols=[' class="text-center"'],
        head=['Arizona Diamondbacks']),
    table(
        clazz='table-fixed border border-bottom-0',
        bcols=[' class="text-center"'],
        body=[
            [
                replace('Previous game', 'Previous game not found') + ' - ' +
                replace('Next game', 'Next game not found')
            ],
        ]),
    table(
        clazz='table-fixed border show-toggler',
        bcols=[' class="text-center"'],
        body=[
            [show('schedule-t31', 'Toggle full schedule')],
        ]),
    table(
        clazz='table-fixed border collapse',
        id_='schedule-t31',
        bcols=[' class="text-center"'],
        body=[
            [span(['text-secondary'], '10/09/2022 @ Los Angeles Dodgers')],
            [anchor('/gameday/1000/', '10/10/2022 @ Los Angeles Dodgers')],
            [anchor('/gameday/2000/', '10/11/2022 @ Los Angeles Dodgers')],
        ]),
    table(
        clazz='table-fixed border border-bottom-0 mt-3',
        hcols=[' class="text-center"'],
        head=['Los Angeles Dodgers']),
    table(
        clazz='table-fixed border border-bottom-0',
        bcols=[' class="text-center"'],
        body=[
            [
                replace('Previous game', 'Previous game not found') + ' - ' +
                replace('Next game', 'Next game not found')
            ],
        ]),
    table(
        clazz='table-fixed border show-toggler',
        bcols=[' class="text-center"'],
        body=[
            [show('schedule-t45', 'Toggle full schedule')],
        ]),
    table(
        clazz='table-fixed border collapse',
        id_='schedule-t45',
        bcols=[' class="text-center"'],
        body=[
            [span(['text-secondary'], '10/09/2022 v Arizona Diamondbacks')],
            [anchor('/gameday/1000/', '10/10/2022 v Arizona Diamondbacks')],
            [anchor('/gameday/2000/', '10/11/2022 v Arizona Diamondbacks')],
        ]),
]

subtitle = 'Diamondbacks at Dodgers, 10/09/2022'

tmpl = 'game.html'

context = {
    'title': 'game',
    'breadcrumbs': _breadcrumbs,
    'raw': _raw,
    'schedule': _schedule
}
