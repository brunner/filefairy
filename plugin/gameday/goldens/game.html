<!doctype html>
<html lang="en">

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link rel="stylesheet"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
      crossorigin="anonymous">
<link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/open-iconic/1.1.1/font/css/open-iconic-bootstrap.min.css">
<title>Fairylab | gameday Â» Diamondbacks at Dodgers, 10/09/2022</title>

<style>
html {
  position: relative;
  min-height: 100%;
}
body {
  margin-bottom: 60px;
}
header {
  background-color: #f5f5f5;
  width: 100%;
}
footer {
  background-color: #f5f5f5;
  bottom: 0;
  height: 60px;
  line-height: 60px;
  position: absolute;
  width: 100%;
}
a.disabled{
  pointer-events: none;
  cursor: default;
}
.alert, .dropdown-toggle, .form-control {
  padding: .75rem;
}
.alert .oi {
  line-height: 24px;
  padding-right: .75rem;
}
.anchor {
  padding-top: 0px;
  margin-top: 0px;
  transition: padding-top .35s ease, margin-top .35s ease;
}
.breadcrumb {
  background-color: inherit;
  line-height: 60px;
  max-width: 468px;
  overflow-x: auto;
}
@media (max-width: 560px) {
  .breadcrumb {
    max-width: 428px;
  }
}
@media (max-width: 512px) {
  .breadcrumb {
    max-width: 380px;
  }
}
@media (max-width: 464px) {
  .breadcrumb {
    max-width: 332px;
  }
}
@media (max-width: 416px) {
  .breadcrumb {
    max-width: 284px;
  }
}
@media (max-width: 368px) {
  .breadcrumb {
    max-width: 236px;
  }
}
.dropdown-toggle {
  color: #6c767c;
}
.flex-grow {
  flex-grow: 1;
}
.left-8p {
  left: 8px;
}
.navbar {
  padding: 0;
}
.navbar-nav {
  padding: .5rem 0;
}
.navbar .nav-link {
  padding: .5rem;
  cursor: pointer;
}
.pl-24p {
  padding-left: 24px;
}
.pr-24p {
  padding-right: 24px;
}
.right-8p {
  right: 8px;
}
.table, .table-sm {
  margin-bottom: 0;
  word-break: break-word;
}
.table .thead-light td {
  background-color: #f5f5f5;
}
.table-fixed {
  table-layout: fixed;
}
.table-sm td {
  border-top: 0;
  padding: 0 .3rem 0 0;
}
.table .td-sm {
  padding: .75rem .25rem;
}
.table.collapse.show {
  display: table;
}
.top-14p {
  top: 14px;
}
.w-10 {
  width: 10%;
}
.w-20 {
  width: 20%;
}
.w-40 {
  width: 40%;
}
.w-55p {
  width: 55px;
}
.w-65p {
  width: 65px;
}
.w-75p {
  width: 75px;
}
.w-105p {
  width: 105px;
}
</style>
</head>

<body>

<!-- ^(base.header) -->
<header class="border-bottom sticky-top">
<div class="container">
<nav class="navbar navbar-light" aria-label="nav">
<ol class="breadcrumb flex-nowrap text-truncate m-0 p-0">
<li class="breadcrumb-item">
<a href="/">Fairylab</a>
</li>
<li class="breadcrumb-item">
<a href="/gameday/">Gameday</a>
</li>
<li class="breadcrumb-item active" aria-current="page">Diamondbacks at Dodgers, 10/09/2022</li>
</ol>

<!-- ^(common.nav) -->
<button class="navbar-toggler border-0" type="button" data-toggle="collapse"
        data-target="#nav-dropdown" aria-controls="#nav-dropdown"
        aria-expanded="false" aria-label="nav toggler">
<span class="navbar-toggler-icon"></span>
</button>
<div class="collapse navbar-collapse border-top" id="nav-dropdown">
<ul class="navbar-nav">
<li class="nav-item">
<span class="nav-link"
      data-toggle="modal"
      data-target="#schedule">Schedule</a>
</li>
</ul>
</div>
<!-- $(common.nav) -->

</nav>
</div>
</header>
<!-- $(base.header) -->

<!-- ^(base.main) -->
<main role="main" class="pb-3">

<!-- ^(game.tabs) -->
<div class="container">

<!-- ^(game.row) -->
<div class="row">

<!-- ^(game.column) -->
<div class="col-12">

<ul class="nav nav-tabs mt-3" id="game-tab" role="tablist">

<li class="nav-item">

<a class="nav-link active" id="raw-tab" data-toggle="tab" href="#raw"
   role="tab" aria-controls="raw" aria-selected="true">Game Log</a>

</li>

</ul>

<div class="tab-content" id="game-tab-content">

<div class="tab-pane fade show active" id="raw"
     role="tabpanel" aria-labelledby="raw-tab">

<!-- ^(common.table) -->
<table class="table border mt-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" colspan="2">Arizona Diamondbacks batting - Pitching for Los Angeles Dodgers : LHP 101</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="w-50">Batting: SHB 102</td>
<td class="w-50">0-0: Ball</td>
</tr>
<tr>
<td scope="row" class="w-50"></td>
<td class="w-50">103 to second</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

<!-- ^(common.table.foot) -->
<tfoot class="thead-light">
<tr>
<td scope="col" colspan="2">0 run(s), 0 hit(s), 0 error(s), 0 left on base</td>
</tr>
</tfoot>
<!-- $(common.table.foot) -->

</table>
<!-- $(common.table) -->

</div>

</div>

</div>
<!-- $(game.column) -->

</div>
<!-- $(game.row) -->

</div>
<!-- $(game.tabs) -->

<div class="modal fade"
     id="schedule"
     tabindex="-1"
     role="dialog"
     aria-labelledby="schedule-modal-label" aria-hidden="true">

<!-- ^(common.modal.head) -->
<div class="modal-dialog" role="document">
<div class="modal-content">
<div class="modal-header">
<h5 class="modal-title" id="schedule-modal-label">Schedule</h5>
<button type="button" class="close" data-dismiss="modal" aria-label="Close">
<span aria-hidden="true">&times;</span>
</button>
</div>
<!-- $(common.modal.head) -->

<!-- ^(common.modal.body) -->
<div class="modal-body">

<!-- ^(common.table) -->
<table class="table table-fixed border border-bottom-0">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="text-center">Arizona Diamondbacks</td>
</tr>
</thead>
<!-- $(common.table.head) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border border-bottom-0">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="text-center"><a class="repl-toggler" href="#" role="button" data-repl="Previous game not found">Previous game</a> - <a class="repl-toggler" href="#" role="button" data-repl="Next game not found">Next game</a></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border show-toggler">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="text-center"><a data-toggle="collapse" href="#schedule-t31" role="button" aria-expanded="false" aria-controls="schedule-t31">Toggle full schedule</a></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border collapse"id="schedule-t31">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="text-center"><span class="text-secondary">10/09/2022 @ Los Angeles Dodgers</span></td>
</tr>
<tr>
<td scope="row" class="text-center"><a href="/gameday/1000/">10/10/2022 @ Los Angeles Dodgers</a></td>
</tr>
<tr>
<td scope="row" class="text-center"><a href="/gameday/2000/">10/11/2022 @ Los Angeles Dodgers</a></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border border-bottom-0 mt-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="text-center">Los Angeles Dodgers</td>
</tr>
</thead>
<!-- $(common.table.head) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border border-bottom-0">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="text-center"><a class="repl-toggler" href="#" role="button" data-repl="Previous game not found">Previous game</a> - <a class="repl-toggler" href="#" role="button" data-repl="Next game not found">Next game</a></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border show-toggler">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="text-center"><a data-toggle="collapse" href="#schedule-t45" role="button" aria-expanded="false" aria-controls="schedule-t45">Toggle full schedule</a></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="table table-fixed border collapse"id="schedule-t45">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="text-center"><span class="text-secondary">10/09/2022 v Arizona Diamondbacks</span></td>
</tr>
<tr>
<td scope="row" class="text-center"><a href="/gameday/1000/">10/10/2022 v Arizona Diamondbacks</a></td>
</tr>
<tr>
<td scope="row" class="text-center"><a href="/gameday/2000/">10/11/2022 v Arizona Diamondbacks</a></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(common.modal.body) -->

</div>

</main>
<!-- $(base.main) -->

<!-- ^(base.footer) -->
<footer class="border-top">
<div class="container text-center">
<small class="text-muted">Last reloaded: 06:02:30 PDT (1985-10-26)</small>
</div>
</footer>
<!-- $(base.footer) -->

<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js"
        integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4"
        crossorigin="anonymous"></script>
<script>
$('.navbar-toggler').one('click', function(e) {
  setTimeout(function() {
    var height = $('header').height() + 1;
    var css = '<style> .anchor.offset { padding-top: %spx; margin-top: -%spx; } </style>';
    $(css.replace(/%s/g, height)).appendTo('head');
  }, 500);
});
$('.navbar-toggler').on('click', function(e) {
  $('.anchor').toggleClass('offset');
});
$('.repl-toggler').on('click', function(e) {
  if ($(this).hasClass('disabled')) {
    e.preventDefault();
  } else {
    $(this).toggleClass('disabled text-secondary');
    $(this).html($(this).data('repl'));
  }
});
$('.show-toggler').on('click', function(e) {
  $(this).toggleClass('border-bottom-0');
});
</script>


</body>

</html>